### DISCUSSION TITLE
Provide multiple texture support for texture progressbar.

### DISCUSSION THREAD

**Discussion by stdgn** [godotengine/godot-proposals#4536] [GUI]
![643](https://user-images.githubusercontent.com/39684712/168292659-69f77f78-2994-4585-8f46-01fb8efd5bfe.jpg)

I have a warehouse in the game. the filled part appears green. the yellow area increases as we try to fill it. As far as I know, there is no such feature in the texture. If such a feature comes, it will reduce the workload.

**Comment by lentsius-bark** [comment]
I actually think this is a great idea, sure you could get this one using custom logic, but a node would make something along these lines much easier, plus it's a relatively common feature in games.

**Comment by miv391** [comment]
Isn't this pretty easy to achieve using two progress bars using same coordinates and same size? Set the foreground color of the top progress bar to green and background color to transparent. Then set the bottom progress bar's foreground color to yellow and disable showing the percentage.

**Reply by Zireael07** [reply]
Often colored progress bar use more than 2 colors (a common case is green-yellow-red or even green-yellow-orange-red). Drawing 3 or 4 progressbars with transparency isn't good for performance...

**Reply by stdgn** [reply]
https://user-images.githubusercontent.com/39684712/168429010-adca0569-6994-49e0-a64f-b9527179005a.mp4

Yes it is, but it would be better to do it with a single texture progress bar. On the contrary, if I want to reduce it, the work will still be prolonged.

**Reply by miv391** [reply]
> Often colored progress bar use more than 2 colors (a common case is green-yellow-red or even green-yellow-orange-red).

I may be missing something here, but I don't think progress bars with multiple values are not common cases at all. If you are talking about progress bars with multiple stages (e.g. 0-30 is green, 30-70 is yellow and 70-100 is red), then `TextureProgress` can already do that. Crude example picture follows:

![kuva](https://user-images.githubusercontent.com/49998025/168430572-e43c8332-13b8-4231-95e5-3c444fdacbc4.png)

**Reply by Calinou** [reply]
> Often colored progress bar use more than 2 colors (a common case is green-yellow-red or even green-yellow-orange-red). Drawing 3 or 4 progressbars with transparency isn't good for performance...

Drawing 4 overlaying ProgressBars or TextureProgress nodes isn't going to affect performance significantly. Also, transparency doesn't affect rendering performance in 2D.

**Reply by stdgn** [reply]
> 4 bindirmeli ProgressBars veya TextureProgress düğümü çizmek, performansı önemli ölçüde etkilemeyecektir. Ayrıca şeffaflık, 2B'de işleme performansını etkilemez.

Does this answer mean that this feature is not coming?

**Reply by Calinou** [reply]
> Does this answer mean that this feature is not coming?

As it stands, this feature is very unlikely to be implemented. Even if it was, you'd have to wait for a new stable Godot release before you can use the feature in production. Therefore, I recommend looking at implementing this yourself using multiple ProgressBars 🙂
